pkg/vsphere/extraconfig/doc.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extraconfig
+
+/*
+Package extraconfig provides Encode/Decode methods to convert data between Go structs and VMware Extraconfig values.append
+
+The implementation understands the following set of annotations and map the fields to appropriate extraConfig keys - in the case where the key describes a boolean state, ommitting the annotation implies the opposite:
+
+hidden - hidden from GuestOS
+read-only - value can only be modified via vSphere APIs
+read-write - value can be modified
+non-persistent - value will be lost on VM reboot
+volatile - field is not exported directly, but via a function that freshens the value each time)
+
+The struct fields are required to be annotated with the "vic" tag, otherwise extraconfig package simply skips them. Scope and key tags are also required.append
```

Comment:
Is the end supposed to be `.append`? If so, what is `required.append`? Same with line 18